libraries/AP_Notify/Buzzer.cpp

```diff
@@ -208,7 +217,210 @@ void Buzzer::on(bool turn_on)
     _flags.on = turn_on;
 
     // pull pin high or low
-    hal.gpio->write(BUZZER_PIN, _flags.on);
+    hal.gpio->write(BUZZER_PIN, _flags.on ? HAL_BUZZER_ON : HAL_BUZZER_OFF);
+}
+
+/// play_pattern - plays the defined buzzer pattern
+void Buzzer::play_pattern(BuzzerPattern pattern_id)
+{
+    _pattern = pattern_id;
+    _pattern_counter = 0;
+}
+
+#else
+
+// buzzer pin as parameter
+
+bool Buzzer::init()
+{
```

Comment:
why have two implementations of init() and update() for with and without BUZZER_PIN? If it is just setting the pin then just init a pin variable in one place and you are done.